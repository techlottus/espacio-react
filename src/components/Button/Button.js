

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


