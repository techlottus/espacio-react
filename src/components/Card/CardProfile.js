import React, { useEffect } from "react";

export const CardProfile = React.memo(({ data }) => {
  const cardRef = React.createRef();

  useEffect(() => {
    cardRef.current.data = {
      image: data.image || "",
      name: data.name || "",
      items: data.items || [],
      disabled: data.disabled,
    };
  }, [data]);

  return (
    <>
      <lottus-card-profile ref={cardRef}></lottus-card-profile>
    </>
  );
});
