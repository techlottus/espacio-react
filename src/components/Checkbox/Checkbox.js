import React, { useEffect } from "react";



export const Checkbox = React.memo(({ data, onCheck }) => {
  const checkboxRef = React.createRef();

  useEffect(() => {
    checkboxRef.current.data = {
      label: data.label || "",
      disabled: data.disabled,
      selected: data.selected,
      name: data.name || "",
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    checkboxRef.current.addEventListener("onCheck", onCheck);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-checkbox ref={checkboxRef}></lottus-checkbox>
    </>
  );
});
