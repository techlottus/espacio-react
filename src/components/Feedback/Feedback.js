import React, { useEffect } from 'react';
import "./Feedback.scss";

export const Feedback = ({data, onRight, text, html=() => {return}, isNoti=false}) => {

    const feedbackRef = React.createRef();
  
    useEffect(() => {
      feedbackRef.current.data = {
        left: data.left || {},
        right: data.right || {},
        type: data.type || '',
        isTextEvent: data.isTextEvent,
        textEvent: data.textEvent || '',
      };
      
    },[data,text]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
      feedbackRef.current.addEventListener('onRight', onRight)
    },[]) // eslint-disable-line react-hooks/exhaustive-deps
    
    return (
      <>
        <lottus-feedback ref={feedbackRef}> 
          <div content="" className={`feedback-content ${isNoti ? "feedbackcontentmessagetext" : ""}`}>{text || html()}</div>
        </lottus-feedback>
      </>
    )
  }
