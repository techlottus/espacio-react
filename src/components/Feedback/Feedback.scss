@import "../../App.scss";

.feedback-content {
  font-family: "Nunito", sans-serif;
  line-height: pixeltorem(19px);
  font-size: pixeltorem(14px);

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    font-size: pixeltorem(16px);
    line-height: pixeltorem(22px);
  }
}

.feedbackcontentmessagetext {
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2; /* number of lines to show */
  line-clamp: 2;
  -webkit-box-orient: vertical;
}