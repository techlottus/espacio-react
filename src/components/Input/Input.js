import React, { useEffect } from 'react'

export const Input = React.memo(({data,value,hasError,errorMessage ,eventFocus,eventKeyPress, eventBlurPress, listenIcon})  => {
    const inputRef = React.createRef();

    useEffect(() => {
        inputRef.current.info = {
            label: data.label || '',
            name: data.name || '',
            type: data.type || '',
            typeButton: data.typeButton || '',
            maxlength: data.maxlength || '',
            onPaste: data.onPaste,
            placeholder: data.placeholder || '',
            autocomplete: data.autocomplete || '',
            disabled: data.disabled,
            alphanumeric: data.alphanumeric,
            alphabetical: data.alphabetical,
            onlyNumbers: data.onlyNumbers,
            upperCase: data.upperCase,
            pattern: data.pattern || '',
            mask: data.mask || ""
        }
    },[data]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
        inputRef.current.value = value;
    },[value]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
        inputRef.current.hasError = hasError;
        inputRef.current.errorMessage = errorMessage;
    },[hasError,errorMessage]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {

        inputRef.current.addEventListener('eventFocus', eventFocus);
        inputRef.current.addEventListener('eventKeyPress', eventKeyPress);
        inputRef.current.addEventListener('eventBlurPress', eventBlurPress);
        inputRef.current.addEventListener('listenIcon', listenIcon);
    }, []); // eslint-disable-line react-hooks/exhaustive-deps

    return(
        <>
        <lottus-input ref={inputRef}></lottus-input>
        </>
    )
})