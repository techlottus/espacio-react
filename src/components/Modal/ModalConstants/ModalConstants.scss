@import '../../../App.scss';
@import "../Modal.scss";

.modal-constants {
  @extend .modal-container;
  align-items: inherit;
  .modalcontent {
    height: 80vh;
    margin-top: 5vh;
    overflow-y: auto;
    @media only screen and (min-width: map-get($grid-breakpoints, s)) {
      width: calc(60% - #{pixeltorem(48px)});
      height: 80vh;
    }
    .modalheader {
      display: flex;
      justify-content: space-between;
    }
    .modalbtns {
      margin-bottom: pixeltorem(16px);
    }
  }
}
