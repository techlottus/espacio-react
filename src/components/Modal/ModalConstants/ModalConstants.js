import React, { useEffect, useRef, useState } from "react";
import { Button } from "../../Button/Button";
import { constantsModalBtn } from "../../../constants/Procedures.constant";
import "./ModalConstants.scss";
import { CardTypeConstant } from "../../Card/CardTypeConstant";
import { useSelector } from "react-redux";
import { downloadFile } from "../../../helpers/download";
import { notiObs } from "../../../observables/notificationObs";
import { typesNoti } from "../../../types/typeNoti";
import env from "../../../enviroment/environment";
import { sendInfoTM } from "../../../tagging/services/sendInfoTagManager";
import {
  closeRecordTypeStudyRecordTag,
  downloadRecordTypeStudyRecordTag,
} from "../../../tagging/flows/procedures/studyRecordTag";

export const ModalConstants = ({ isShow, onClose, HandleDemo, outside }) => {
  const { modal } = useSelector((state) => state.procedureStudyRecord);
  const { texts } = useSelector((state) => state.texts);

  const modalRef = useRef();

  const [isModal, setIsModal] = useState(false); 

  useEffect(() => {
    setIsModal(isShow);
  }, [isShow]);

  const [desktop, setDesktop] = useState(window.innerWidth > 991);
  const updateMedia = () => {
    setDesktop(window.innerWidth > 991);
  };

  useEffect(() => {
    window.addEventListener("resize", updateMedia);
    return () => window.removeEventListener("resize", updateMedia);
  });
  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target) &&
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const handleModal = () => {
    return (
      <div className="modal-constants">
        <div className="modalcontent" ref={modalRef}>
          <div className="modalheader">
            <div className="title">
              {texts?.procedures?.modalConstants?.title}
            </div>
            <span
              className="material-icons icon"
              onClick={() => {
                onClose();
                sendInfoTM(
                  window,
                  closeRecordTypeStudyRecordTag,
                  "modal_t_close"
                );
              }}
            >
              close
            </span>
          </div>
          <div className="btnconstants">
            <div className="btnprimary">
              <Button
                data={{
                  ...constantsModalBtn,
                  isExpand: window.innerWidth < 991,
                }}
                onClick={() => {
                  downloadFile(
                    "tipos-constancias.pdf",
                    `${env.contentAssets}docs/constancia/utc/tipos-constancias.pdf`
                  )
                    .then(() => {
                      notiObs.next({
                        type: typesNoti.success,
                        text: texts.procedures.downloadPDFSuccess,
                      });
                      sendInfoTM(
                        window,
                        downloadRecordTypeStudyRecordTag,
                        "modal_t_demo"
                      );
                    })
                    .catch(() => {
                      notiObs.next({
                        type: typesNoti.error,
                        text: texts.procedures.downloadPDFError,
                      });
                    });
                  onClose();
                }}
              />
            </div>
          </div>
          <div className="modalbody">
            {modal.map((dataModal, i) => {
              const data = {
                title: dataModal.label,
                text: dataModal.description,
                image: desktop ? dataModal.imageUrl : null,
              };
              return <CardTypeConstant data={data} key={i} />;
            })}
          </div>
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
