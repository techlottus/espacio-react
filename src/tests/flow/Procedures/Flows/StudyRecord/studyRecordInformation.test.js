import React from "react";
import { mount } from "enzyme";
import { Provider} from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { proceduresText } from "../../../../../texts/proceduresText";
import StudyRecordInformation from "../../../../../flow/Procedures/Flows/StudyRecord/StudyRecordInformation";
import { BrowserRouter } from "react-router-dom";
import { ProceduresContext } from "../../../../../flow/Procedures/ProceduresContext";
import { dataStudyRecordWihtoutInfoMock } from "../../../../../fixtures/procedures/studyRecordFixture";
import { infoProceduresMock } from "../../../../../fixtures/procedures/procedureFixture";
import { Input } from "../../../../../components/Input/Input";
import { act } from "react-dom/test-utils";
import { CardProfile } from "../../../../../components/Card/CardProfile";
import { Select } from "../../../../../components/Select/Select";
import { tokenWithDelivaryMock } from "../../../../../fixtures/tokenFixture";
import { Checkbox } from "../../../../../components/Checkbox/Checkbox";
import { getValuesOfAuth } from "../../../../../helpers/auth";
import { Feedback } from "../../../../../components/Feedback/Feedback";
import { Button } from "../../../../../components/Button/Button";
import { ModalConstants } from "../../../../../components/Modal/ModalConstants/ModalConstants";
//import { useFormBuilder } from "../../../../../hooks/useForm";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

jest.mock('../../../../../hooks/useForm',() => ({
  useFormBuilder: () => {
    return [
      {
        get: () => {
          return {
            setValue: () => {}
          }
        }
      },
      jest.fn().mockImplementation(() => {}),
      {
        value: {},
        valid: false
      },
      (nuewState) => {},
      {},
      jest.fn().mockImplementation(() => {}),
    ]
  }
}))

jest.mock('../../../../../helpers/auth',() => ({
  getValuesOfAuth: () => {
    return {
      modality: 'withDelivery'
    }
  }
}));

let initState = {
  texts: {
    procedures: {
      ...proceduresText,
    },
  },
  login: {
    accessToken: tokenWithDelivaryMock
  },
  procedureStudyRecord: {
    ...dataStudyRecordWihtoutInfoMock
  },
  procedures: {
    ...infoProceduresMock
  }
};

window.scrollTo = jest.fn()
const handleSteps = jest.fn().mockImplementation(() => {});

const handleMount = (initState) => {
  let store = mockStore(initState);
  return mount(
    <ProceduresContext.Provider value={{handleSteps}}>
      <Provider store={store}>
        <BrowserRouter>
          <StudyRecordInformation />
        </BrowserRouter>
      </Provider>
    </ProceduresContext.Provider>
  );
};

describe("Test <StudyRecordInformation />", () => {

  beforeEach(() => {
    
  }) 


  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });

  test("should not render the profile ", () => {
    const initWithoutProfile = {
      ...initState,
      procedures: {
        ...initState.procedures,
        profile: null
      }
    }
    const wrapper = handleMount(initWithoutProfile);
    expect(wrapper.find(CardProfile).exists()).toBe(false);
  });

  test("should form valid with campus ", () => {
    initState = {
      ...initState,
      procedureStudyRecord: {
        ...initState.procedureStudyRecord,
        data: {
          ...initState.procedureStudyRecord.data,
          selectTypeDelivery: '1'
        }
      }
    }
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Input).first().props().eventKeyPress({
        detail: {
          value:'5512345678'
        }
      })
      wrapper.find(Select).forEach((select,index) => {
        select.props().onClick({
          detail: '2'
        })
      })
    })
    //jest.spyOn(React, "useState").mockImplementation([{},() => {}]);
    wrapper.update();
    act(() => {
      wrapper.find(Button).last().props().onClick();
    })
    expect(wrapper).toMatchSnapshot();
    
  });

  test("should form valid without campus ", () => {
   
   
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Input).first().props().eventKeyPress({
        detail: {
          value:'5512345678'
        }
      })
      wrapper.find(Select).forEach((select,index) => {
        select.props().onClick({
          detail: '1'
        })
      })
      wrapper.find(Checkbox).props().onCheck({
        detail: true
      })
      wrapper.find(Input).last().props().eventKeyPress({
        detail: {
          value:'Hola Mundo'
        }
      })
    })
    expect(wrapper).toMatchSnapshot();
  });

  test('should back main procedures',() => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Button).first().props().onClick();
    })
  })

  test('should open study record notice',() => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Feedback).find('.studyrecordnoticelink').props().onClick();
    })
  })

  test('should close study record notice',() => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(ModalConstants).props().onClose();
    })
  })

  describe('should back of documentation',() => {
    test('wiht compus',() => {
      initState = {
        ...initState,
        procedureStudyRecord: {
          ...initState.procedureStudyRecord,
          data: {
            ...initState.procedureStudyRecord.data,
            phone: '5512345678',
            selectTypeRecord: '1',
            selectTypeDelivery: '1',
            selectCampusDelivery: '1'
          }
        }
      }
      const wrapper = handleMount(initState);
      
    })

    test('without campus',() => {
      initState = {
        ...initState,
        procedureStudyRecord: {
          ...initState.procedureStudyRecord,
          data: {
            ...initState.procedureStudyRecord.data,
            phone: '5512345678',
            selectTypeRecord: '1',
            selectTypeDelivery: '2',
          }
        }
      }
      const wrapper = handleMount(initState);
     
    })
  })

  test('should not found profile',() => {
    initState = {
      ...initState,
      procedureStudyRecord: {
        ...initState.procedureStudyRecord,
        information: {
          ...initState.procedureStudyRecord.information,
          profile: null
        }
      }
    }
    const wrapper = handleMount(initState);
    expect(wrapper.find(CardProfile).exists()).toBe(false);
  })

});
