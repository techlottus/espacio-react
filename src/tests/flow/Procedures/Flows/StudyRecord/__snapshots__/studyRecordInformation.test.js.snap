// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <StudyRecordInformation /> should form valid with campus  1`] = `
Array [
  <div
    className="studyrecordinfoprofile"
  >
    <lottus-card-profile />
  </div>,
  <div
    className="studyrecordinfonumber"
  >
    <p
      className="studyrecordinfonumbertitle"
    />
    <div
      className="studyrecordinfoinput"
    >
      <lottus-input />
    </div>
  </div>,
  <div
    className="studyrecordnotice"
  >
    <lottus-feedback>
      <div
        className="feedback-content"
        content=""
      >
        <div>
          Conoce los
          <span
            className="studyrecordnoticelink"
            onClick={[Function]}
          >
            tipos de constacia
          </span>
          según tus necesidades
        </div>
      </div>
    </lottus-feedback>
  </div>,
  <div
    className="studyrecordform"
  >
    <div
      className="stydyrecordformselect"
    >
      <lottus-select />
    </div>
    <div
      className="stydyrecordformselect"
    >
      <lottus-select />
    </div>
    <div
      className="stydyrecordformcheck"
    >
      <lottus-checkbox />
    </div>
    <div
      className="stydyrecordformitem"
    >
      <lottus-input />
    </div>
  </div>,
  <div
    className="studyrecordbtns"
  >
    <div
      className="studyrecordbtn"
    >
      <lottus-button />
    </div>
    <div
      className="studyrecordbtn"
    >
      <lottus-button />
    </div>
  </div>,
  "",
]
`;

exports[`Test <StudyRecordInformation /> should form valid without campus  1`] = `
Array [
  <div
    className="studyrecordinfoprofile"
  >
    <lottus-card-profile />
  </div>,
  <div
    className="studyrecordinfonumber"
  >
    <p
      className="studyrecordinfonumbertitle"
    />
    <div
      className="studyrecordinfoinput"
    >
      <lottus-input />
    </div>
  </div>,
  <div
    className="studyrecordnotice"
  >
    <lottus-feedback>
      <div
        className="feedback-content"
        content=""
      >
        <div>
          Conoce los
          <span
            className="studyrecordnoticelink"
            onClick={[Function]}
          >
            tipos de constacia
          </span>
          según tus necesidades
        </div>
      </div>
    </lottus-feedback>
  </div>,
  <div
    className="studyrecordform"
  >
    <div
      className="stydyrecordformselect"
    >
      <lottus-select />
    </div>
    <div
      className="stydyrecordformselect"
    >
      <lottus-select />
    </div>
    <div
      className="stydyrecordformselect"
    >
      <lottus-select />
    </div>
    <div
      className="stydyrecordformcheck"
    >
      <lottus-checkbox />
    </div>
    <div
      className="stydyrecordformitem"
    >
      <lottus-input />
    </div>
  </div>,
  <div
    className="studyrecordbtns"
  >
    <div
      className="studyrecordbtn"
    >
      <lottus-button />
    </div>
    <div
      className="studyrecordbtn"
    >
      <lottus-button />
    </div>
  </div>,
  "",
]
`;

exports[`Test <StudyRecordInformation /> should init component 1`] = `
Array [
  <div
    className="studyrecordinfoprofile"
  >
    <lottus-card-profile />
  </div>,
  <div
    className="studyrecordinfonumber"
  >
    <p
      className="studyrecordinfonumbertitle"
    />
    <div
      className="studyrecordinfoinput"
    >
      <lottus-input />
    </div>
  </div>,
  <div
    className="studyrecordnotice"
  >
    <lottus-feedback>
      <div
        className="feedback-content"
        content=""
      >
        <div>
          Conoce los
          <span
            className="studyrecordnoticelink"
            onClick={[Function]}
          >
            tipos de constacia
          </span>
          según tus necesidades
        </div>
      </div>
    </lottus-feedback>
  </div>,
  <div
    className="studyrecordform"
  >
    <div
      className="stydyrecordformselect"
    >
      <lottus-select />
    </div>
    <div
      className="stydyrecordformselect"
    >
      <lottus-select />
    </div>
    <div
      className="stydyrecordformcheck"
    >
      <lottus-checkbox />
    </div>
    <div
      className="stydyrecordformitem"
    >
      <lottus-input />
    </div>
  </div>,
  <div
    className="studyrecordbtns"
  >
    <div
      className="studyrecordbtn"
    >
      <lottus-button />
    </div>
    <div
      className="studyrecordbtn"
    >
      <lottus-button />
    </div>
  </div>,
  "",
]
`;
